src/basic/log.c

```diff
@@ -686,9 +686,8 @@ int log_internalv_realm(
         if (_likely_(LOG_PRI(level) > log_max_level[realm]))
                 return -error;
 
-        /* Make sure that %m maps to the specified error */
-        if (error != 0)
-                errno = error;
+        /* Make sure that %m maps to the specified error (or "Success"). */
+        errno = error;
```

Comment:
hmm, i'd think we should add one tweak here, and change this to:

```c
if (error != INT_MIN)
        errno = error;
```

And then change the definitions of log_full() from this:

```c
#define log_full(level, ...) log_full_errno((level), 0, __VA_ARGS__)
```

to this:

```c
#define log_full(level, ...) log_full_errno((level), INT_MIN, __VA_ARGS__)
```
i.e. the log_xyz() calls that do not take any error code should pass INT_MIN when wrapping the full parameter version, so that we can then filter that out again.

Why bother? Well, otherwise doing `log_error("foobar: %m")` would consistently print "foobar: Success" but I think to maintain traditional printf() %m behaviour in this case makes sense, simply because no error is passed at all.

Or to say this differently: let's just shift the "best of both worlds" approach a bit: instead of enabling the printf() mode by error == 0 let's do so with error == INT_MIN. 

Why INT_MIN? well, any other special value that is clearly recognizable and outside of the typical error number range would work too. I think INT_MIN is kinda nice though, as error numbers the way strerror() takes them are >= 0. Most of our codes when seeing negative values will just do abs() around it though. But, on 2's complement systems that's not going to work for INT_MIN, and hence it's kinda nice to have as special indicator for "use system errno", as -INT_MIN is impossible to map to any reasonable error code...

Reply:
hmm, when we do this, the `if (error < 0) error = -error` line further up needs some changing I guess...

Reply:
ok